### PR TITLE
lint: Enable linter for "pkg/virtcl/credentials" package

### PR DESCRIPTION
**Pull request by akrejcir**
### What this PR does
Enabled linter for `pkg/virtcl/credentials` package and fixed issues found.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/virtctl/credentials/addkey/addkey.go [hunk 1]
```diff
@@ -10,7 +10,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/tools/clientcmd"
-	"k8s.io/utils/pointer"
+	"k8s.io/utils/ptr"
```
Comment (EdDev) [inline pkg/virtctl/credentials/addkey/addkey.go:13]: Please use "kubevirt.io/kubevirt/pkg/pointer". This dependency is redundant.

File: pkg/virtctl/credentials/addkey/addkey.go [hunk 2]
```diff
@@ -100,25 +103,31 @@ func runAddKeyCommand(clientConfig clientcmd.ClientConfig, cmdFlags *addSshKeyFl
...
-	return updateSecretWithSshKey(cmd, cli, cmdFlags, vm, sshKey)
+	return updateSecretWithSSHKey(cmd, cli, cmdFlags, vm, sshKey)
 }
 
-func addSecretWithSshKey(cmd *cobra.Command, cli kubecli.KubevirtClient, cmdFlags *addSshKeyFlags, vm *v1.VirtualMachine, sshKey string) (err error) {
+func addSecretWithSSHKey(cmd *cobra.Command,
+	cli kubecli.KubevirtClient,
+	cmdFlags *addSSHKeyFlags,
+	vm *v1.VirtualMachine,
+	sshKey string) error {
```
Comment (EdDev) [inline pkg/virtctl/credentials/addkey/addkey.go:115]: I am unsure to which format pattern you follow when breaking the lines.

This comment is general, not only related to this incident.

I know of these format patterns:
The trivial `foo(a, b ,c)`.
The next best thing:
```
foo(
    a, b, c,
)
```

and the one-per-line one:
```
foo(
    a,
    b,
    c,
)
```

In this case I would go with the last, e.g.
```
func addSecretWithSSHKey(
	cmd *cobra.Command,
	cli kubecli.KubevirtClient,
	cmdFlags *addSSHKeyFlags,
	vm *v1.VirtualMachine,
	sshKey string,
) error {
```

As we do not have a formatter that enforces this automatically, it is not a must.
But I have hard time following the arguments when it is not well organized.

File: pkg/virtctl/credentials/addkey/addkey_test.go [hunk 3]
```diff
@@ -21,7 +21,7 @@ import (
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/kubernetes/fake"
 	k8stesting "k8s.io/client-go/testing"
-	"k8s.io/utils/pointer"
+	"k8s.io/utils/ptr"
```
Comment (EdDev) [inline pkg/virtctl/credentials/addkey/addkey_test.go:24]: Please use "kubevirt.io/kubevirt/pkg/pointer".

File: pkg/virtctl/credentials/addkey/addkey_test.go [hunk 4]
```diff
@@ -291,7 +294,7 @@ var _ = Describe("Credentials add-ssh-key", func() {
 		const testKey = "key contents in file"
 
 		filename := filepath.Join(GinkgoT().TempDir(), "test-key-file")
-		Expect(os.WriteFile(filename, []byte(testKey), 0666)).To(Succeed())
+		Expect(os.WriteFile(filename, []byte(testKey), 0600)).To(Succeed())
```
Comment (EdDev) [inline pkg/virtctl/credentials/addkey/addkey_test.go:297]: How is this change related to the linter?

File: pkg/virtctl/credentials/removekey/removekey_test.go [hunk 5]
```diff
@@ -20,7 +21,7 @@ import (
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/kubernetes/fake"
 	k8stesting "k8s.io/client-go/testing"
-	"k8s.io/utils/pointer"
+	"k8s.io/utils/ptr"
```
Comment (EdDev) [inline pkg/virtctl/credentials/removekey/removekey_test.go:24]: Please use "kubevirt.io/kubevirt/pkg/pointer".

File: pkg/virtctl/credentials/removekey/removekey_test.go [hunk 6]
```diff
@@ -239,7 +242,7 @@ var _ = Describe("Credentials", func() {
 
 	It("should remove key from secret with key from file", func() {
 		filename := filepath.Join(GinkgoT().TempDir(), "test-key-file")
-		Expect(os.WriteFile(filename, []byte(testKey), 0666)).To(Succeed())
+		Expect(os.WriteFile(filename, []byte(testKey), 0600)).To(Succeed())
```
Comment (EdDev) [inline pkg/virtctl/credentials/removekey/removekey_test.go:245]: How is this related to the linter coverage?

File: pkg/virtctl/credentials/password/password.go [hunk 7]
```diff
@@ -88,9 +88,9 @@ func runSetPasswordCommand(clientConfig clientcmd.ClientConfig, cmdFlags *passwo
 	}
 
 	if !cmdFlags.Force {
-		secret, err := cli.CoreV1().Secrets(vm.Namespace).Get(cmd.Context(), secretName, metav1.GetOptions{})
-		if err != nil {
-			return fmt.Errorf("error getting secret \"%s\": %w", secretName, err)
+		secret, err2 := cli.CoreV1().Secrets(vm.Namespace).Get(cmd.Context(), secretName, metav1.GetOptions{})
```
Comment (EdDev) [inline pkg/virtctl/credentials/password/password.go:91]: Please use a more meaningful name for the error, like `errs` or `errSecret`.

File: pkg/virtctl/credentials/password/password_test.go [hunk 8]
```diff
@@ -1,18 +1,18 @@
...
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/utils/pointer"
 
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/utils/ptr"
```
Comment (EdDev) [inline pkg/virtctl/credentials/password/password_test.go:15]: Please use "kubevirt.io/kubevirt/pkg/pointer".

**Review by EdDev — COMMENTED** [review]
Nice cleanup, thank you!

/sig code-quality

File: pkg/virtctl/credentials/addkey/addkey_test.go (hunk 4 above)
Comment (akrejcir) [reply pkg/virtctl/credentials/addkey/addkey_test.go:297]: It was checked by the linter. I got this error message:
```
pkg/virtctl/credentials/addkey/addkey_test.go:297:10: G306: Expect WriteFile permissions to be 0600 or less (gosec)
		Expect(os.WriteFile(filename, []byte(testKey), 0666)).To(Succeed())
		       ^
```

Comment (EdDev) [reply pkg/virtctl/credentials/addkey/addkey_test.go:297]: Interesting. Thanks for sharing.
Perhaps you can add it to the commit message, it is not that trivial.

File: pkg/virtctl/credentials/BUILD.bazel [hunk 9]
```diff
@@ -6,9 +6,9 @@ go_library(
     importpath = "kubevirt.io/kubevirt/pkg/virtctl/credentials",
     visibility = ["//visibility:public"],
     deps = [
-        "//pkg/virtctl/credentials/add-key:go_default_library",
-        "//pkg/virtctl/credentials/remove-key:go_default_library",
-        "//pkg/virtctl/credentials/set-password:go_default_library",
+        "//pkg/virtctl/credentials/addkey:go_default_library",
+        "//pkg/virtctl/credentials/password:go_default_library",
+        "//pkg/virtctl/credentials/removekey:go_default_library",
```
Comment (codingben) [inline pkg/virtctl/credentials/BUILD.bazel:11]: Why do you need to have `addkey` and `removekey`? I'd suggest to just have `pkg/virtctl/credentials/sshkey`, wdyt?

File: pkg/virtctl/credentials/addkey/addkey.go (hunk 1 above)
Comment (akrejcir) [reply pkg/virtctl/credentials/addkey/addkey.go]: done

File: pkg/virtctl/credentials/removekey/removekey_test.go (hunk 5 above)
Comment (akrejcir) [reply pkg/virtctl/credentials/removekey/removekey_test.go]: done

File: pkg/virtctl/credentials/addkey/addkey_test.go (hunk 3 above)
Comment (akrejcir) [reply pkg/virtctl/credentials/addkey/addkey_test.go]: done

File: pkg/virtctl/credentials/addkey/addkey_test.go (hunk 4 above)
Comment (akrejcir) [reply pkg/virtctl/credentials/addkey/addkey_test.go:297]: done

File: pkg/virtctl/credentials/removekey/removekey_test.go (hunk 6 above)
Comment (akrejcir) [reply pkg/virtctl/credentials/removekey/removekey_test.go:245]: done

File: pkg/virtctl/credentials/password/password.go (hunk 7 above)
Comment (akrejcir) [reply pkg/virtctl/credentials/password/password.go]: done

File: pkg/virtctl/credentials/password/password_test.go (hunk 8 above)
Comment (akrejcir) [reply pkg/virtctl/credentials/password/password_test.go]: done

File: pkg/virtctl/credentials/BUILD.bazel (hunk 9 above)
Comment (akrejcir) [reply pkg/virtctl/credentials/BUILD.bazel:11]: Each of these packages contains a single `virtctl` command. We could merge them to a single package, but I'd prefer it to be in a different PR. This PR only fixes linter issues.

**Review by EdDev — APPROVED** [review]

**Review by enp0s3 — COMMENTED** [review]
/approve
/hold
@akrejcir Thank you!
@xpivarc It looks good to me, not sure if `PatchSet` could be used here as an improvement, feel free to unhold if you think its OK

**Comment by EdDev** [comment]
> @xpivarc It looks good to me, not sure if `PatchSet` could be used here as an improvement, feel free to unhold if you think its OK

@enp0s3 could you please clarify how PatchSet is related to the linter coverage?
Is this related to some other comment in the PR?

**Comment by enp0s3** [comment]
@EdDev Ack. It actually doesn't. We can improve the code by using the `PatchSet` in a follow-up

**Comment by brianmcarey** [comment]
/hold

The lint lane is failing - @akrejcir I think you might need to rebase.

**Comment by akrejcir** [comment]
@brianmcarey , I've rebased and the lint is passing. Can you unhold?

**Comment by brianmcarey** [comment]
/hold cancel

thanks @akrejcir !

**Review by 0xFelix — APPROVED** [review]
/lgtm

Thanks
